omod-2.2/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/ObsController2_2Test.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs2_2;
+
+import org.junit.Test;
+import org.openmrs.Obs;
+import org.openmrs.api.ObsService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.RestTestConstants1_8;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
```

Comment:
Is there any reason why you did not add this test to the existing `ObsController1_9Test`?

Reply:
YES @dkayiwa due to
>public Obs getRevisionObs(Obs initialObs);

	/**
	 * Gets the value of status currently saved in the database for the given obs, bypassing any caches. This is used
	 * when the user updates an existing obs so we can determine whether to change its status or not.
	 * @param obs
	 * @return
	 * @since 2.1.0
	 */

Which works since 2.1.0

Reply:
Why do you need that function?

Reply:
Since it returns the updated obs .....@ibacher

>public Obs getObsByUuid(String uuid) throws APIException;

	/**
	 * Get Revision Obs for initial Obs
	 *
	 * @param initialObs
	 * @return obs or null
	 * @since 2.1
	 * <strong>Should</strong> find revision obs for given valid obs
	 * <strong>Should</strong> return null if no revision obs found for given obs
	 */

can I use this up? ........have found it also helpful

Reply:
@ibacher this method returns the updated obs....
but have also seen 
>@Authorized(PrivilegeConstants.GET_OBS)
	public Obs getObsByUuid(String uuid) throws APIException;

	/**
	 * Get Revision Obs for initial Obs
	 *
	 * @param initialObs
	 * @return obs or null
	 * @since 2.1
	 * <strong>Should</strong> find revision obs for given valid obs
	 * <strong>Should</strong> return null if no revision obs found for given obs
	 */
Can I use this ....